oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/OneandoneComputeServiceAdapter.java

```diff
@@ -83,56 +85,112 @@
       Hardware hardware = template.getHardware();
       TemplateOptions options = template.getOptions();
       Server updateServer = null;
+      PasswordGenerator generate = new PasswordGenerator();
+      Types.ServerType serverType = Types.ServerType.cloud;
 
       final String loginUser = isNullOrEmpty(options.getLoginUser()) ? "root" : options.getLoginUser();
-      final String password = options.hasLoginPassword() ? options.getLoginPassword() : passwordGenerator.generate();
+      final String password = options.hasLoginPassword() ? options.getLoginPassword() : generate.generate();
       final String privateKey = options.hasLoginPrivateKey() ? options.getPrivateKey() : null;
       final org.jclouds.compute.domain.Image image = template.getImage();
       final int[] inboundPorts = template.getOptions().getInboundPorts();
-
-      //prepare hdds to provision
-      List<? extends Volume> volumes = hardware.getVolumes();
-      List<Hdd.CreateHdd> hdds = new ArrayList<Hdd.CreateHdd>();
-
-      for (final Volume volume : volumes) {
-         try {
-            //check if the bootable device has enough size to run the appliance(image).
-            float minHddSize = volume.getSize();
-            if (volume.isBootDevice()) {
-               SingleServerAppliance appliance = api.serverApplianceApi().get(image.getId());
-               if (appliance.minHddSize() > volume.getSize()) {
-                  minHddSize = appliance.minHddSize();
-               }
-            }
-            Hdd.CreateHdd hdd = Hdd.CreateHdd.create(minHddSize, volume.isBootDevice());
-            hdds.add(hdd);
-         } catch (Exception ex) {
-            throw Throwables.propagate(ex);
-
+      String imageId = image.getId();
+      HardwareFlavour hardwareModel = isFlavor(hardware.getId());
+      boolean isBaremetal = hardware.getName() == null ? false : hardware.getName().contains(baremetalModelsKey);
```

Comment:
I think worth introducing a `OneandoneTemplateOptions extends TemplateOptions` to deal with specific requirements a user may specify like bare metal or even `hardwareModel` above. wdyt?

Reply:
My intention was to allow the users to just pass in the flavor name i.e. `BMC_XL` or `XL` as a `hardwareId` to choose between baremetal and cloud servers.

Maybe the `OneandoneTemplateOptions` would be an overkill since the only thing users will need to provide is the different hardware flavor name which can be achieved using the `public TemplateBuilder hardwareId(String string);` from the `TeamplateBuilder` class